### PR TITLE
[SERVICES-2592] Portfolio queries cache improvements

### PR DESCRIPTION
**Pull request by mad2sm0key**
## Reasoning
- `userTokens` and `userNfts` queries perform a lot of single GET requests to Redis  
  
## Proposed Changes
- identify Promise.all() calls that perform Redis requests and replace them with batch get requests
  - add bulk getter method for farm token IDs
  - small refactor on Pair dataloaders
  - bulk getter for price discovery : redeem token IDs
  - bulk getters for simple lock : lock token ids, lp token ids, farm proxy token ids
  - bulk getter for staking proxy : dual yield token IDs
  - bulk getter for staking abi : farm token IDs
- compute user fungible tokens USD value using bulk methods

## How to test
- N/A

### CODE REVIEW

File: src/modules/simple-lock/services/simple.lock.abi.service.ts [hunk 1]
```diff
@@ -39,6 +44,18 @@ export class SimpleLockAbiService
         return response.firstValue.valueOf().toString();
     }
 
+    async getAllLockedTokenIds(
```
Comment (claudiulataretu) [inline src/modules/simple-lock/services/simple.lock.abi.service.ts:47]: I don't think it's worth having this type of methods in here. We only have 3 simple lock contracts

File: src/modules/user/services/user.esdt.service.ts [hunk 2]
```diff
@@ -11,88 +10,65 @@ import { TokenService } from 'src/modules/tokens/services/token.service';
...
-    private async getEsdtTokenDetails(token: EsdtToken): Promise<UserToken> {
-        const pairAddress = await this.pairService.getPairAddressByLpTokenID(
-            token.identifier,
-        );
+        const userLpEsdtTokens = userTokens
+            .filter((token) => lpTokensIDs.includes(token.identifier))
+            .map((token) => new EsdtToken(token));
+
+        const orderedPairAddresses = [];
+        userLpEsdtTokens.forEach((token, index) => {
```
Comment (claudiulataretu) [inline src/modules/user/services/user.esdt.service.ts:55]: this `forEach` is hard to understand why you need a list of ordered pair addresses

**Review by claudiulataretu — CHANGES_REQUESTED** [review]

File: src/modules/user/services/user.esdt.service.ts (hunk 2 above)
Comment (mad2sm0key) [reply src/modules/user/services/user.esdt.service.ts:55]: The call `this.userEsdtCompute.allLpTokensUSD(..)` accepts an array of esdt tokens, and an array of pair addresses that need to match the LP tokens.

File: src/modules/simple-lock/services/simple.lock.abi.service.ts (hunk 1 above)
Comment (mad2sm0key) [reply src/modules/simple-lock/services/simple.lock.abi.service.ts]: removed all 3 methods

**Review by cfaur09 — DISMISSED** [review]

**Review by claudiulataretu — DISMISSED** [review]

**Review by claudiulataretu — APPROVED** [review]

**Review by cfaur09 — APPROVED** [review]

**Pull request merged**